### PR TITLE
Japron fresh

### PR DESCRIPTION
**Pull request by tianhan0**
Apron environment is set up in this pull request.

An Apron test case is included.

### CODE REVIEW

File: src/test/scala/edu/colorado/plv/cuanto/numdomain/NumDomainSpec.scala [hunk 1]
```diff
@@ -0,0 +1,140 @@
+package edu.colorado.plv.cuanto.numdomain
+
+import apron.{Box, Manager, Octagon, Polka, _}
+import gmp.Mpfr
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.util.{Try, Failure, Success}
+
+/**
+  * Created by Tianhan Lu on 4/20/17.
```
Comment (bechang) [inline src/test/scala/edu/colorado/plv/cuanto/numdomain/NumDomainSpec.scala]: Nit, but please follow what you agreed to: https://cuplv.slack.com/archives/C3S635ZD4/p1497045800481119

**Comment by octalsrc** [comment]
Running the `numdomain.*` tests a second time in the same `sbt` session fails with this error:

```
test


Box
null
full: <universal>
empty: <empty>
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fc427c05734, pid=2597, tid=0x00007fc3bf8b6700
#
# JRE version: OpenJDK Runtime Environment (8.0_131) (build 1.8.0_131-11)
# Java VM: OpenJDK 64-Bit Server VM (25.131-b11 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# V  [libjvm.so+0x40f734]  CompilationPolicy::can_be_compiled(methodHandle, int)+0x154
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/octal/cuanto/hs_err_pid2597.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
/nix/store/vk0njs96zdsbp40fi40m8fq1569jwyxf-sbt-0.13.15/share/sbt/bin/sbt-launch-lib.bash: line 46:  2597 Aborted                 "$@"
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -70,6 +70,27 @@ manager, you will need to find and install the following tools:
 - JDK 8
 - `sbt`
 - `z3`
+- `gmp`
+- `mpfr`
+
+After installing the tools listed above, you need to download Apron source code,
```
Comment (bechang) [inline CONTRIBUTING.md]: Let's list here just the Apron dependency (link to the project or something).

Then, move this to a section on Apron installation.

Comment (bechang) [reply CONTRIBUTING.md]: Actually if there are linux versus mac specific parts, then put those sections in the Mac/Linux installation sections, respectively

File: src/test/scala/edu/colorado/plv/cuanto/numdomain/NumDomainSpec.scala [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  */
+class NumDomainSpec extends FlatSpec with Matchers {
+  runAPITest()
```
Comment (bechang) [inline src/test/scala/edu/colorado/plv/cuanto/numdomain/NumDomainSpec.scala]: Please put this in a ScalaTest spec

File: src/test/scala/edu/colorado/plv/cuanto/numdomain/NumDomainSpec.scala [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    /* level 0 abstract elements */
+    val full = new Abstract0(man, 2, 1)
+    val empty = new Abstract0(man, 2, 1, true)
+    val a0 = new Abstract0(man, 2, 1, box)
+    println("full: " + full)
+    println("empty: " + empty)
+    println("a0: " + a0)
+    println("a0: " + a0.toString(man, names))
+    assert(!full.isBottom(man))
```
Comment (bechang) [inline src/test/scala/edu/colorado/plv/cuanto/numdomain/NumDomainSpec.scala]: Shouldn't these asserts be in separate tests?

File: src/test/scala/edu/colorado/plv/cuanto/numdomain/NumDomainSpec.scala [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+import apron.{Box, Manager, Octagon, Polka, _}
+import gmp.Mpfr
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.util.{Try, Failure, Success}
+
+/**
+  * Created by Tianhan Lu on 4/20/17.
+  */
+class NumDomainSpec extends FlatSpec with Matchers {
```
Comment (bechang) [inline src/test/scala/edu/colorado/plv/cuanto/numdomain/NumDomainSpec.scala]: Gut this class and put it into JapronWorksheet.sc.

**Comment by tianhan0** [comment]
1. The worksheet stays in `edu.colorado.plv.cuanto.numdomain`
2. Write a `XXXSpec.scala` in order to test a few Apron APIs (to ensure it runs on someone's machine without crashing)

File: src/test/scala/edu/colorado/plv/cuanto/numdomain/JapronWorksheet.sc [hunk 6]
```diff
@@ -0,0 +1,140 @@
+package edu.colorado.plv.cuanto.numdomain
```
Comment (bechang) [inline src/test/scala/edu/colorado/plv/cuanto/numdomain/JapronWorksheet.sc:1]: Be able to run this (in IntelliJ).

Comment (bechang) [reply src/test/scala/edu/colorado/plv/cuanto/numdomain/JapronWorksheet.sc:1]: Try editing the library path via SBT.

**Comment by tianhan0** [comment]
SBT will look for user contributing libraries in some directories?

**Comment by tianhan0** [comment]
Define an abstract domain in jsy, and use japron to operate on that abstract domain.

Start by looking at test cases on illustrating how to use the jsy language

**Comment by bechang** [comment]
So this is ready to review? It doesn't have WIP on it.

**Comment by bechang** [comment]
@octalsrc is the reviewer?

**Review by octalsrc — APPROVED** [review]
Looks good.  This PR performs the work to make Japron available to the Cuanto code-base and tests that it is indeed available.

**Pull request merged**